src/s2_fos/training/open_ai_prompts.py

```diff
@@ -0,0 +1,209 @@
+import numpy as np
+
+from s2_fos.src import FOS_LIST
+FOS_LIST = [
+    "Agricultural and Food sciences",
+    "Art",
+    "Biology",
+    "Business",
+    "Chemistry",
+    "Computer science",
+    "Economics",
+    "Education",
+    "Engineering",
+    "Environmental science",
+    "Geography",
+    "Geology",
+    "History",
+    "Law",
+    "Linguistics",
+    "Materials science",
+    "Mathematics",
+    "Medicine",
+    "Philosophy",
+    "Physics",
+    "Political science",
+    "Psychology",
+    "Sociology",
+    'Other'
+]
+
+
+def search_for_fos_respons(response):
+    fields = []
+    for field in FOS_LIST:
+        if field in response:
+            fields.append(field)
+
+    return fields
+
+
+def structure_open_ai_response(open_ai_calls):
+    open_ai_list = []
+    for key, value in open_ai_calls.items():
+        paper_sha = key[:40]
+        model = key[41:]
+        # print(paper_sha, model)
+        extracted_fields = search_for_fos_respons(value['result']['choices'][0]['message']['content'])
+        completion_tokens = value['result']['usage']['completion_tokens']
+        prompt_tokens = value['result']['usage']['prompt_tokens']
+        open_ai_list.append({'paper_sha': paper_sha, 'model': model,
+                             'FoS_openAI': extracted_fields, 'completion_tokens': completion_tokens,
+                             'prompt_tokens': prompt_tokens})
+
+    return open_ai_list
+
+
+from pyspark.sql.types import StructType, StructField, StringType, ArrayType, IntegerType
+
+
+def create_spark_df_open_ai(open_ai_calls):
+    open_ai_calls_dict_list = structure_open_ai_response(open_ai_calls)
+    schema = StructType([
+        StructField("paper_sha", StringType(), True),
+        StructField("model", StringType(), True),
+        StructField("FoS_openAI", ArrayType(StringType()), True),
+        StructField("completion_tokens", IntegerType(), True),
+        StructField("prompt_tokens", IntegerType(), True),
+    ])
+    return spark.createDataFrame(open_ai_calls_dict_list, schema=schema)
+
+
+import re
+
+def tokenize(text):
+    if text:
+        return re.findall(r'\b\w+\b', text)
+    return text
+
+def truncate_entity(entity, max_tokens=800):
+    tokens = tokenize(entity)
+    if tokens and len(tokens) > max_tokens:
+        tokens = tokens[:max_tokens]
+        truncated_entity = ' '.join(tokens)
+        return truncated_entity
+    else:
+        return entity
+
+
+import openai
```

Comment:
It makes sense to document this but if we release the data, the user will never need to spend all that money, right? Does this code rerun along with the "retraining" instructions in the readme?

Reply:
No it does not, my thinking was to add the functional part of the code to call OpenAI with the prompt, in case users are interested in getting more data or modify the prompt.

Reply:
This [code](https://github.com/allenai/s2_fos/blob/5b4ee54698477bdece63d2b019e86424d60e8ae8/src/s2_fos/training/open_ai_prompts.py) runs and collects/saves open_ai calls into json format. I've added in README.md instructions on how to run it.

I will add openAI returns to the hugging face repository along with training data.